File: testdata/payloads/fork_bomb/bomb.py
# fork_bomb: tests pids.max limit and process creation exhaustion
import os
import sys

def bomb():
    try:
        while True:
            os.fork()
    except Exception as e:
        sys.stderr.write(f"Fork failed safely: {e}\n")
        sys.exit(1)

if __name__ == "__main__":
    bomb()

File: testdata/payloads/fork_bomb/bomb.cpp
// fork_bomb: tests pids.max limit and process creation exhaustion
#include <unistd.h>
#include <iostream>

int main() {
    while (true) {
        pid_t p = fork();
        if (p < 0) {
            std::cerr << "Fork failed due to pids limit\n";
            return 1;
        }
    }
    return 0;
}
